syne_tune/backend/sagemaker_backend/sagemaker_utils.py

```diff
@@ -446,3 +447,25 @@ def delete_action(s3_client, bucket: str, object_key: str) -> Optional[str]:
     return _s3_traverse_recursively(
         s3_client=s3, action=delete_action, bucket=bucket_name, prefix=prefix
     )
+
+
+def set_backend_path_not_synced_to_s3(trial_backend: TrialBackend):
+    """
+    When an experiment with the local backend is run remotely (as SageMaker
+    training job), we do not want checkpoints to be synced to S3, since this
+    is expensive and error-prone (since several trials may write checkpoints
+    at the same time).
+
+    Here, we direct checkpoint writing to /opt/ml/input/data/, which is mounted
+    on a partition with sufficient space. Different to /opt/ml/checkpoints, this
+    directory is not synced to S3.
+
+    .. note::
+       This has to be called *after* the :class:`~syne_tune.Tuner` object has
+       been created, because the latter sets the backend path as well.
+
+    :param trial_backend: Backend in question
+    """
+    path = Path("/opt/ml/input/data/")
+    path.mkdir(parents=True, exist_ok=True)
+    trial_backend.set_path(results_root=str(path))
```

Comment:
Just to make sure, this does not change the path for results.csv.zip and metadata.json , right?

Reply:
No, it does not. It is a bit tricky. When `tuner` gets constructed, it both sets `tuner.tuner_path`, where tuning results are written, but also it sets `tuner.trial_backend.set_path(...)` to the same path, and this is where the backend writes info per trial, such as checkpoints and logs.

Here, only the latter is changed.

I tested this, results are written as before.

Reply:
And there is also a reason why we are not writing just to "~/", or something returned by `tempfile`. It seems there is not so much space there for SM training jobs, but there is more space at "/opt/ml/input/data/". The documentation on this stuff is non-existing, though. You should run it with a large Hugging Face model and check that things work.

Reply:
BTW: When `delete_checkpoints=True` for the backend (I think this is default), then we remove checkpoints once a trial is stopped or terminates (but NOT if it is paused). This can save some space when checkpoints are large.

On the other hand, for ASHA promotion, most trials are paused until the end, so it does not help much.

Reply:
You should check what happens for PBT: Are we removing checkpoints which are no longer needed? If not, we should re-think this, because our checkpoints will only ever get larger.